### PR TITLE
416 IRS plan cleanup

### PR DESCRIPTION
**Pull request by cKellyDesign**
For #416 - This PR is a starting point for IRS cleanup and adding tests

- Removes New Plan functionality
- Removes Edit Plan Name functionality
- Adds missing docstrings
- Removes `isNewPlan` state / functionality
- Removes unused imports and variables
- Fixes typos (`jurisidction` and `decendant`)
- Updates all method docstrings to include @params with typings
- Consolidates fill color settings
- Replaces some `any`s with more specific typings

### CODE REVIEW

**Comment by cKellyDesign** [comment]
This PR is ready for review, it's just incremental progress towards #416 and is largely subtractive.

File: src/containers/pages/InterventionPlan/IRS/plan/tests/index.test.tsx [hunk 1]
```diff
@@ -62,4 +64,31 @@ describe('containers/pages/IRS/plan', () => {
     expect(toJson(wrapper)).toMatchSnapshot();
     wrapper.unmount();
   });
+
+  // it('loads planById from Plans store', () => {});
```
Comment (moshthepitt) [inline src/containers/pages/InterventionPlan/IRS/plan/tests/index.test.tsx]: Could we remove this commented out code?

**Comment by cKellyDesign** [comment]
> should we refactor to not use componentWillReceiveProps now instead of waiting for the (near) future?

Now might be a good time, since this is more refactoring than cleanup I would think this PR isn't the place.

**Comment by cKellyDesign** [comment]
@moshthepitt - there are still some `any`s in this module, mostly around the following areas which I think may warrant their own tickets/PRs:

1. Gisida / Mapbox specific objects which don't have explicit types (gisida) or are missing properties (mapbox)
2. MouseEvents, for some reason the default MouseEvents and React synthetic events aren't playing nicely, I found this when trying to type them a while ago and don't remember the specific issue

**Comment by moshthepitt** [comment]
> > should we refactor to not use componentWillReceiveProps now instead of waiting for the (near) future?
> 
> Now might be a good time, since this is more refactoring than cleanup I would think this PR isn't the place.

Do you think we should create an issue for this?

**Comment by cKellyDesign** [comment]
>>> should we refactor to not use componentWillReceiveProps now instead of waiting for the (near) future?
>>
>> Now might be a good time, since this is more refactoring than cleanup I would think this PR isn't the place.
>
> Do you think we should create an issue for this?

@moshthepitt ya we have one here https://github.com/onaio/reveal-frontend/issues/237

**Pull request merged**